### PR TITLE
add moloch

### PR DESCRIPTION
**Pull request by jamesyoung**

### CODE REVIEW

File: src/contracts/service/Fraction.sol [hunk 1]
```diff
@@ -4,22 +4,65 @@ pragma solidity ^0.7.0;
...
     /**
      * @param _nftids - An array of NFT ids
      */
     function fungify(ERC721 _nft, uint256[] memory _nftids) public virtual {
         require(locked == false);
 
+        address[] memory approvedTokens;
+        approvedTokens[0] = address(erc20Token);
+
+        dao = new Moloch(
```
Comment (raymondfeng) [inline src/contracts/service/Fraction.sol:55]: Should we create the DAO as a separate operation as `fungify()` can be potentially called multiple times to add new NFTs?

At least you should check if the DAO has already created.

File: src/contracts/service/CollabLandDAO.sol [hunk 2]
```diff
@@ -3,10 +3,10 @@
...
 
 import "moloch/contracts/Moloch.sol";
-import "moloch/contracts/v3/core/DaoFactory.sol";
-import "moloch/contracts/v3/core/proposals/Proposal.sol";
-import "moloch/contracts/v3/core/membership/Member.sol";
-import "moloch/contracts/v3/adapters/Voting.sol";
+// import "moloch/contracts/v3/core/DaoFactory.sol";
+// import "moloch/contracts/v3/core/proposals/Proposal.sol";
+// import "moloch/contracts/v3/core/membership/Member.sol";
+// import "moloch/contracts/v3/adapters/Voting.sol";
```
Comment (raymondfeng) [inline src/contracts/service/CollabLandDAO.sol:9]: I added these import statements as a workaround to force typechain to generate corresponding TypeScript classes/interaces. I guess we can move it to a dummy contract.

File: src/contracts/service/BondingCurve.sol [hunk 3]
```diff
@@ -12,6 +12,8 @@ contract BondingCurve {
     using SafeMath for uint256;
     address payable public owner;
 
+    bool public halted;
```
Comment (raymondfeng) [inline src/contracts/service/BondingCurve.sol:15]: Is it used by other contracts yet?

File: src/contracts/service/Fraction.sol [hunk 4]
```diff
@@ -4,22 +4,65 @@ pragma solidity ^0.7.0;
...
+        processingReward = _processingReward;
+    }
 
     /**
      * @param _nftids - An array of NFT ids
      */
     function fungify(ERC721 _nft, uint256[] memory _nftids) public virtual {
         require(locked == false);
 
+        address[] memory approvedTokens;
```
Comment (raymondfeng) [inline src/contracts/service/Fraction.sol:52]: When do we set `locked` to true?

File: src/contracts/service/Fraction.sol (hunk 1 above)
Comment (jamesyoung) [reply src/contracts/service/Fraction.sol:55]: great point, I think we should only allow fungify to be called once.

File: src/contracts/service/BondingCurve.sol (hunk 3 above)
Comment (jamesyoung) [reply src/contracts/service/BondingCurve.sol:15]: I was going to use it for buyout, will remove

File: src/contracts/service/Fraction.sol (hunk 4 above)
Comment (jamesyoung) [reply src/contracts/service/Fraction.sol:52]: it should be called after fungify has been called, making fungify only executed once

**Pull request merged**